repaygithub/cactus
# CACTUS-347 :: Storyshots

Ticket: https://repayonline.atlassian.net/browse/CACTUS-347

### A Few Notes
- I had to downgrade Jest to v24.0.1 to get the storyshots to work...it was either that or upgrade to storybook v6.
- `Spinner` isn't being tested by the storyshots because it's always moving so I don't think there's a way we can get the snapshots to match consistently on that one.
- There's a `test:visual` yarn script in the `cactus-web` directory, but I wouldn't recommend running this command outside of Docker; the storyshots should be run in a Docker container so they're always run on the same OS and therefore won't have any variations with text sizing and such.
- They are intended to be run from the root using the `test:build` and `test:docker` commands.
- I haven't the faintest clue why, but the docs website tests were failing because of a type that Gatsby defines. I added `--skipLibCheck` to get around this for now, but we should probably figure out why that is at some point.


### Testing
1. Pull down the branch and run `test:build`. Wait for Docker to finish building.
2. Run `test:docker` and make sure all of the snapshots pass.
